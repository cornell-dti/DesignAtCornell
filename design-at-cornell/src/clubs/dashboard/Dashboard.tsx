import React from 'react';
import { VerticalFlex } from '../../components/ContainerStyles';
import { Grid } from '../../components/DashboardGridStyles';
import ClubBubble from './ClubBubble';
import { Club } from '../../../../server/types';

const Dashboard = (clubs: Club[]) => {
  const clubBubbles = (
    <Grid>
      {Object.values(clubs).map((club) => (
        <ClubBubble key={club.title} {...club} />
      ))}
    </Grid>
  );

  return (
    <VerticalFlex>
      <br></br>
      {clubBubbles}
    </VerticalFlex>
  );
};

export default Dashboard;
