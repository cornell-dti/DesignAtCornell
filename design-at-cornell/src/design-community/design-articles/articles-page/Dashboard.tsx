import React from 'react';
import { HorizontalFlex, VerticalFlex } from '../../../components/ContainerStyles';
import { ArticleGrid } from '../../../components/DashboardGridStyles';
import { HeadingLine } from '../../../components/DashboardElementStyles';
import { Article } from '../../../../../server/types';
import ArticleBubble from './Bubble';
import LargeArticleBubble from './LargeBubble';
const Dashboard = (article: Article[]) => {
  const articleBubbles = (
    <ArticleGrid>
      {Object.values(article)
        .slice(0, 1)
        .map((article) => (
          <LargeArticleBubble key={article.id} {...article} />
        ))}
      {Object.values(article)
        .slice(1)
        .map((article) => (
          <ArticleBubble key={article.id} {...article} />
        ))}
    </ArticleGrid>
  );

  return (
    <VerticalFlex>
      <br></br>
      <br></br>
      <HorizontalFlex>
        <HeadingLine>
          <hr />
          <h3>Articles</h3>
          <line />
        </HeadingLine>
      </HorizontalFlex>
      <br></br>
      <br></br>
      {articleBubbles}
      <br></br>
    </VerticalFlex>
  );
};

export default Dashboard;
