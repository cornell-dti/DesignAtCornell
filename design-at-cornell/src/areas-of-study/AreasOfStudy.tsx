import React, { useEffect, useState } from 'react';
import { PageContainer } from './AreasOfStudyStyles';
import Title from './title/Title';
import Dashboard from './dashboard/Dashboard';
import axios from 'axios';
import { Major } from '../../../server/types';

export type AreaOfStudy = {
  name: string;
  description: string;
};

export type Studies = {
  readonly majors: ReadonlyArray<Major>;
  readonly minors: ReadonlyArray<Major>;
  readonly gradStudies: ReadonlyArray<Major>;
}

const AreasOfStudy = () => {

  const [majors, setMajors] = useState<ReadonlyArray<Major>>([]);

  const majorsURL = 'http://localhost:3000/getMajors';
  useEffect(() => {
    axios
      .get<{
        success: boolean;
        data: Major[];
      }>(majorsURL)
      .then(res => res.data.data)
      .then(setMajors)
  }, []);

  const studies: Studies = {
    majors: majors,
    minors: majors,
    gradStudies: majors
  }

  return (
    <PageContainer>
      <Title />
      <Dashboard {...studies} />
    </PageContainer>
  );
}

export default AreasOfStudy;
