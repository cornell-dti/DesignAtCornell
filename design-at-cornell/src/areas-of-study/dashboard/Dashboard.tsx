import React from 'react';
import { 
  DashboardContainer, 
  StudiesContainer, 
  Divider
} from '../AreasOfStudyStyles';
import StudiesGrid from './StudiesGrid';
import { Studies } from '../AreasOfStudy';
import ApplyTags from './ApplyTags';

const Dashboard = (studies: Studies) => {
  return (
    <DashboardContainer>
      <StudiesContainer>
        <Divider>
          <h1>Undergraduate Majors</h1>
          <hr />
        </Divider>
        <StudiesGrid {...studies.majors} />
        <Divider>
          <h1>Undergraduate Minors</h1>
          <hr />
        </Divider>
        <StudiesGrid {...studies.minors} />
        <Divider>
          <h1>Graduate Studies</h1>
          <hr />
        </Divider>
        <StudiesGrid {...studies.grad_studies} />
      </StudiesContainer>
      <ApplyTags />
    </DashboardContainer>
  );
}

export default Dashboard;
