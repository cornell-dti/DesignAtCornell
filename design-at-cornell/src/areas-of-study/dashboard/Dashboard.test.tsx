import React from 'react';
import renderer from 'react-test-renderer';
import Dashboard from './Dashboard';
import { Studies } from '../AreasOfStudy';
import 'jest-styled-components';

describe('<Dashboard />', () => {
  it('renders correctly without any majors, minors, grad_studies', () => {
    const studies: Studies = {
      majors: [],
      minors: [],
      grad_studies: [],
    };
    const snapshot = renderer.create(<Dashboard {...studies} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('renders correctly with one area of study in each category', () => {
    const studies: Studies = {
      majors: [{ name: 'major 1', description: 'this is a major' }],
      minors: [{ name: 'minor 1', description: 'this is a minor' }],
      grad_studies: [{ name: 'gread area 1', description: 'this is a grad area' }],
    };
    const snapshot = renderer.create(<Dashboard {...studies} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('renders correctly with multiple areas in each category', () => {
    const studies: Studies = {
      majors: [
        { name: 'major 1', description: 'this is a major' },
        { name: 'major 2', description: 'this is a major' },
      ],
      minors: [
        { name: 'minor 1', description: 'this is a minor' },
        { name: 'minor 2', description: 'this is a minor' },
      ],
      grad_studies: [
        { name: 'gread area 1', description: 'this is a grad area' },
        { name: 'gread area 2', description: 'this is a grad area' },
        { name: 'gread area 3', description: 'this is a grad area' },
      ],
    };
    const snapshot = renderer.create(<Dashboard {...studies} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('renders correctly with only majors', () => {
    const studies: Studies = {
      majors: [
        { name: 'major 1', description: 'this is a major' },
        { name: 'major 2', description: 'this is a major' },
      ],
      minors: [],
      grad_studies: [],
    };
    const snapshot = renderer.create(<Dashboard {...studies} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
