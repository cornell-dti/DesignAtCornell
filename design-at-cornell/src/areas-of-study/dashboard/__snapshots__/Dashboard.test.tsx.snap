// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> renders correctly with multiple areas in each category 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 80%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  padding: 60px 90px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  height: 70px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 h1 {
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  font-weight: bold;
  font-size: 28px;
  color: black;
  margin: 0;
}

.c2 hr {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  height: 0;
  border: 4px solid #8ed663;
  margin-left: 20px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  width: 287px;
  margin-bottom: 38px;
}

.c4 h1 {
  font-weight: bold;
  font-size: 16px;
  color: black;
  cursor: pointer;
}

.c4 p {
  font-size: 13px;
  color: #939393;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 20%;
  height: auto;
  background: rgba(142,214,99,0.17);
  padding: 60px;
}

.c5 h5 {
  font-weight: bold;
  font-size: 20px;
  line-height: 70px;
  color: black;
  margin: 0;
}

.c5 h6 {
  font-weight: bold;
  font-size: 16px;
  line-height: 40px;
  color: black;
  margin: 0;
}

.c5 form {
  font-weight: bold;
  font-size: 12px;
  line-height: 24px;
  color: black;
}

.c5 form input[type='checkbox'] {
  vertical-align: middle;
  margin-right: 10px;
  cursor: pointer;
}

.c5 form input[type='checkbox']:before {
  content: '';
  position: absolute;
  width: 13px;
  height: 13px;
  border-radius: 2px;
  border: 2px solid black;
  background-color: white;
}

.c5 form input[type='checkbox']:checked:after {
  content: '';
  position: absolute;
  width: 13px;
  height: 13px;
  border-radius: 2px;
  border: 2px solid black;
  background-color: black;
}

.c5 form label {
  display: block;
  margin-top: -24px;
  margin-left: 20px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <h1>
        Undergraduate Majors
      </h1>
      <hr />
    </div>
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        <h1>
          major 1
        </h1>
        <p>
          this is a major
        </p>
      </div>
      <div
        className="c4"
      >
        <h1>
          major 2
        </h1>
        <p>
          this is a major
        </p>
      </div>
    </div>
    <div
      className="c2"
    >
      <h1>
        Undergraduate Minors
      </h1>
      <hr />
    </div>
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        <h1>
          minor 1
        </h1>
        <p>
          this is a minor
        </p>
      </div>
      <div
        className="c4"
      >
        <h1>
          minor 2
        </h1>
        <p>
          this is a minor
        </p>
      </div>
    </div>
    <div
      className="c2"
    >
      <h1>
        Graduate Studies
      </h1>
      <hr />
    </div>
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        <h1>
          gread area 1
        </h1>
        <p>
          this is a grad area
        </p>
      </div>
      <div
        className="c4"
      >
        <h1>
          gread area 2
        </h1>
        <p>
          this is a grad area
        </p>
      </div>
      <div
        className="c4"
      >
        <h1>
          gread area 3
        </h1>
        <p>
          this is a grad area
        </p>
      </div>
    </div>
  </div>
  <div
    className="c5"
  >
    <h5>
      Apply Tags
    </h5>
    <h6>
      Design Areas
    </h6>
    <form>
      <div>
        <input
          id="Art + Design"
          type="checkbox"
        />
        <label
          htmlFor="Art + Design"
        >
          Art + Design
        </label>
      </div>
      <div>
        <input
          id="Digital Design"
          type="checkbox"
        />
        <label
          htmlFor="Digital Design"
        >
          Digital Design
        </label>
      </div>
      <div>
        <input
          id="Industrial Design"
          type="checkbox"
        />
        <label
          htmlFor="Industrial Design"
        >
          Industrial Design
        </label>
      </div>
      <div>
        <input
          id="Policy Design"
          type="checkbox"
        />
        <label
          htmlFor="Policy Design"
        >
          Policy Design
        </label>
      </div>
      <div>
        <input
          id="Spatial Design"
          type="checkbox"
        />
        <label
          htmlFor="Spatial Design"
        >
          Spatial Design
        </label>
      </div>
    </form>
    <h6>
      Schools
    </h6>
    <form>
      <div>
        <input
          id="AAP"
          type="checkbox"
        />
        <label
          htmlFor="AAP"
        >
          AAP
        </label>
      </div>
      <div>
        <input
          id="Agriculture and Life Sciences"
          type="checkbox"
        />
        <label
          htmlFor="Agriculture and Life Sciences"
        >
          Agriculture and Life Sciences
        </label>
      </div>
      <div>
        <input
          id="Arts and Sciences"
          type="checkbox"
        />
        <label
          htmlFor="Arts and Sciences"
        >
          Arts and Sciences
        </label>
      </div>
      <div>
        <input
          id="Engineering"
          type="checkbox"
        />
        <label
          htmlFor="Engineering"
        >
          Engineering
        </label>
      </div>
      <div>
        <input
          id="Human Ecology"
          type="checkbox"
        />
        <label
          htmlFor="Human Ecology"
        >
          Human Ecology
        </label>
      </div>
      <div>
        <input
          id="ILR"
          type="checkbox"
        />
        <label
          htmlFor="ILR"
        >
          ILR
        </label>
      </div>
      <div>
        <input
          id="SC Johnson School of Business"
          type="checkbox"
        />
        <label
          htmlFor="SC Johnson School of Business"
        >
          SC Johnson School of Business
        </label>
      </div>
    </form>
  </div>
</div>
`;

exports[`<Dashboard /> renders correctly with one area of study in each category 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 80%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  padding: 60px 90px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  height: 70px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 h1 {
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  font-weight: bold;
  font-size: 28px;
  color: black;
  margin: 0;
}

.c2 hr {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  height: 0;
  border: 4px solid #8ed663;
  margin-left: 20px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  width: 287px;
  margin-bottom: 38px;
}

.c4 h1 {
  font-weight: bold;
  font-size: 16px;
  color: black;
  cursor: pointer;
}

.c4 p {
  font-size: 13px;
  color: #939393;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 20%;
  height: auto;
  background: rgba(142,214,99,0.17);
  padding: 60px;
}

.c5 h5 {
  font-weight: bold;
  font-size: 20px;
  line-height: 70px;
  color: black;
  margin: 0;
}

.c5 h6 {
  font-weight: bold;
  font-size: 16px;
  line-height: 40px;
  color: black;
  margin: 0;
}

.c5 form {
  font-weight: bold;
  font-size: 12px;
  line-height: 24px;
  color: black;
}

.c5 form input[type='checkbox'] {
  vertical-align: middle;
  margin-right: 10px;
  cursor: pointer;
}

.c5 form input[type='checkbox']:before {
  content: '';
  position: absolute;
  width: 13px;
  height: 13px;
  border-radius: 2px;
  border: 2px solid black;
  background-color: white;
}

.c5 form input[type='checkbox']:checked:after {
  content: '';
  position: absolute;
  width: 13px;
  height: 13px;
  border-radius: 2px;
  border: 2px solid black;
  background-color: black;
}

.c5 form label {
  display: block;
  margin-top: -24px;
  margin-left: 20px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <h1>
        Undergraduate Majors
      </h1>
      <hr />
    </div>
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        <h1>
          major 1
        </h1>
        <p>
          this is a major
        </p>
      </div>
    </div>
    <div
      className="c2"
    >
      <h1>
        Undergraduate Minors
      </h1>
      <hr />
    </div>
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        <h1>
          minor 1
        </h1>
        <p>
          this is a minor
        </p>
      </div>
    </div>
    <div
      className="c2"
    >
      <h1>
        Graduate Studies
      </h1>
      <hr />
    </div>
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        <h1>
          gread area 1
        </h1>
        <p>
          this is a grad area
        </p>
      </div>
    </div>
  </div>
  <div
    className="c5"
  >
    <h5>
      Apply Tags
    </h5>
    <h6>
      Design Areas
    </h6>
    <form>
      <div>
        <input
          id="Art + Design"
          type="checkbox"
        />
        <label
          htmlFor="Art + Design"
        >
          Art + Design
        </label>
      </div>
      <div>
        <input
          id="Digital Design"
          type="checkbox"
        />
        <label
          htmlFor="Digital Design"
        >
          Digital Design
        </label>
      </div>
      <div>
        <input
          id="Industrial Design"
          type="checkbox"
        />
        <label
          htmlFor="Industrial Design"
        >
          Industrial Design
        </label>
      </div>
      <div>
        <input
          id="Policy Design"
          type="checkbox"
        />
        <label
          htmlFor="Policy Design"
        >
          Policy Design
        </label>
      </div>
      <div>
        <input
          id="Spatial Design"
          type="checkbox"
        />
        <label
          htmlFor="Spatial Design"
        >
          Spatial Design
        </label>
      </div>
    </form>
    <h6>
      Schools
    </h6>
    <form>
      <div>
        <input
          id="AAP"
          type="checkbox"
        />
        <label
          htmlFor="AAP"
        >
          AAP
        </label>
      </div>
      <div>
        <input
          id="Agriculture and Life Sciences"
          type="checkbox"
        />
        <label
          htmlFor="Agriculture and Life Sciences"
        >
          Agriculture and Life Sciences
        </label>
      </div>
      <div>
        <input
          id="Arts and Sciences"
          type="checkbox"
        />
        <label
          htmlFor="Arts and Sciences"
        >
          Arts and Sciences
        </label>
      </div>
      <div>
        <input
          id="Engineering"
          type="checkbox"
        />
        <label
          htmlFor="Engineering"
        >
          Engineering
        </label>
      </div>
      <div>
        <input
          id="Human Ecology"
          type="checkbox"
        />
        <label
          htmlFor="Human Ecology"
        >
          Human Ecology
        </label>
      </div>
      <div>
        <input
          id="ILR"
          type="checkbox"
        />
        <label
          htmlFor="ILR"
        >
          ILR
        </label>
      </div>
      <div>
        <input
          id="SC Johnson School of Business"
          type="checkbox"
        />
        <label
          htmlFor="SC Johnson School of Business"
        >
          SC Johnson School of Business
        </label>
      </div>
    </form>
  </div>
</div>
`;

exports[`<Dashboard /> renders correctly with only majors 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 80%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  padding: 60px 90px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  height: 70px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 h1 {
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  font-weight: bold;
  font-size: 28px;
  color: black;
  margin: 0;
}

.c2 hr {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  height: 0;
  border: 4px solid #8ed663;
  margin-left: 20px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  width: 287px;
  margin-bottom: 38px;
}

.c4 h1 {
  font-weight: bold;
  font-size: 16px;
  color: black;
  cursor: pointer;
}

.c4 p {
  font-size: 13px;
  color: #939393;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 20%;
  height: auto;
  background: rgba(142,214,99,0.17);
  padding: 60px;
}

.c5 h5 {
  font-weight: bold;
  font-size: 20px;
  line-height: 70px;
  color: black;
  margin: 0;
}

.c5 h6 {
  font-weight: bold;
  font-size: 16px;
  line-height: 40px;
  color: black;
  margin: 0;
}

.c5 form {
  font-weight: bold;
  font-size: 12px;
  line-height: 24px;
  color: black;
}

.c5 form input[type='checkbox'] {
  vertical-align: middle;
  margin-right: 10px;
  cursor: pointer;
}

.c5 form input[type='checkbox']:before {
  content: '';
  position: absolute;
  width: 13px;
  height: 13px;
  border-radius: 2px;
  border: 2px solid black;
  background-color: white;
}

.c5 form input[type='checkbox']:checked:after {
  content: '';
  position: absolute;
  width: 13px;
  height: 13px;
  border-radius: 2px;
  border: 2px solid black;
  background-color: black;
}

.c5 form label {
  display: block;
  margin-top: -24px;
  margin-left: 20px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <h1>
        Undergraduate Majors
      </h1>
      <hr />
    </div>
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        <h1>
          major 1
        </h1>
        <p>
          this is a major
        </p>
      </div>
      <div
        className="c4"
      >
        <h1>
          major 2
        </h1>
        <p>
          this is a major
        </p>
      </div>
    </div>
    <div
      className="c2"
    >
      <h1>
        Undergraduate Minors
      </h1>
      <hr />
    </div>
    <div
      className="c3"
    />
    <div
      className="c2"
    >
      <h1>
        Graduate Studies
      </h1>
      <hr />
    </div>
    <div
      className="c3"
    />
  </div>
  <div
    className="c5"
  >
    <h5>
      Apply Tags
    </h5>
    <h6>
      Design Areas
    </h6>
    <form>
      <div>
        <input
          id="Art + Design"
          type="checkbox"
        />
        <label
          htmlFor="Art + Design"
        >
          Art + Design
        </label>
      </div>
      <div>
        <input
          id="Digital Design"
          type="checkbox"
        />
        <label
          htmlFor="Digital Design"
        >
          Digital Design
        </label>
      </div>
      <div>
        <input
          id="Industrial Design"
          type="checkbox"
        />
        <label
          htmlFor="Industrial Design"
        >
          Industrial Design
        </label>
      </div>
      <div>
        <input
          id="Policy Design"
          type="checkbox"
        />
        <label
          htmlFor="Policy Design"
        >
          Policy Design
        </label>
      </div>
      <div>
        <input
          id="Spatial Design"
          type="checkbox"
        />
        <label
          htmlFor="Spatial Design"
        >
          Spatial Design
        </label>
      </div>
    </form>
    <h6>
      Schools
    </h6>
    <form>
      <div>
        <input
          id="AAP"
          type="checkbox"
        />
        <label
          htmlFor="AAP"
        >
          AAP
        </label>
      </div>
      <div>
        <input
          id="Agriculture and Life Sciences"
          type="checkbox"
        />
        <label
          htmlFor="Agriculture and Life Sciences"
        >
          Agriculture and Life Sciences
        </label>
      </div>
      <div>
        <input
          id="Arts and Sciences"
          type="checkbox"
        />
        <label
          htmlFor="Arts and Sciences"
        >
          Arts and Sciences
        </label>
      </div>
      <div>
        <input
          id="Engineering"
          type="checkbox"
        />
        <label
          htmlFor="Engineering"
        >
          Engineering
        </label>
      </div>
      <div>
        <input
          id="Human Ecology"
          type="checkbox"
        />
        <label
          htmlFor="Human Ecology"
        >
          Human Ecology
        </label>
      </div>
      <div>
        <input
          id="ILR"
          type="checkbox"
        />
        <label
          htmlFor="ILR"
        >
          ILR
        </label>
      </div>
      <div>
        <input
          id="SC Johnson School of Business"
          type="checkbox"
        />
        <label
          htmlFor="SC Johnson School of Business"
        >
          SC Johnson School of Business
        </label>
      </div>
    </form>
  </div>
</div>
`;

exports[`<Dashboard /> renders correctly without any majors, minors, grad_studies 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 80%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  padding: 60px 90px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  height: 70px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 h1 {
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  font-weight: bold;
  font-size: 28px;
  color: black;
  margin: 0;
}

.c2 hr {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  height: 0;
  border: 4px solid #8ed663;
  margin-left: 20px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 20%;
  height: auto;
  background: rgba(142,214,99,0.17);
  padding: 60px;
}

.c4 h5 {
  font-weight: bold;
  font-size: 20px;
  line-height: 70px;
  color: black;
  margin: 0;
}

.c4 h6 {
  font-weight: bold;
  font-size: 16px;
  line-height: 40px;
  color: black;
  margin: 0;
}

.c4 form {
  font-weight: bold;
  font-size: 12px;
  line-height: 24px;
  color: black;
}

.c4 form input[type='checkbox'] {
  vertical-align: middle;
  margin-right: 10px;
  cursor: pointer;
}

.c4 form input[type='checkbox']:before {
  content: '';
  position: absolute;
  width: 13px;
  height: 13px;
  border-radius: 2px;
  border: 2px solid black;
  background-color: white;
}

.c4 form input[type='checkbox']:checked:after {
  content: '';
  position: absolute;
  width: 13px;
  height: 13px;
  border-radius: 2px;
  border: 2px solid black;
  background-color: black;
}

.c4 form label {
  display: block;
  margin-top: -24px;
  margin-left: 20px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <h1>
        Undergraduate Majors
      </h1>
      <hr />
    </div>
    <div
      className="c3"
    />
    <div
      className="c2"
    >
      <h1>
        Undergraduate Minors
      </h1>
      <hr />
    </div>
    <div
      className="c3"
    />
    <div
      className="c2"
    >
      <h1>
        Graduate Studies
      </h1>
      <hr />
    </div>
    <div
      className="c3"
    />
  </div>
  <div
    className="c4"
  >
    <h5>
      Apply Tags
    </h5>
    <h6>
      Design Areas
    </h6>
    <form>
      <div>
        <input
          id="Art + Design"
          type="checkbox"
        />
        <label
          htmlFor="Art + Design"
        >
          Art + Design
        </label>
      </div>
      <div>
        <input
          id="Digital Design"
          type="checkbox"
        />
        <label
          htmlFor="Digital Design"
        >
          Digital Design
        </label>
      </div>
      <div>
        <input
          id="Industrial Design"
          type="checkbox"
        />
        <label
          htmlFor="Industrial Design"
        >
          Industrial Design
        </label>
      </div>
      <div>
        <input
          id="Policy Design"
          type="checkbox"
        />
        <label
          htmlFor="Policy Design"
        >
          Policy Design
        </label>
      </div>
      <div>
        <input
          id="Spatial Design"
          type="checkbox"
        />
        <label
          htmlFor="Spatial Design"
        >
          Spatial Design
        </label>
      </div>
    </form>
    <h6>
      Schools
    </h6>
    <form>
      <div>
        <input
          id="AAP"
          type="checkbox"
        />
        <label
          htmlFor="AAP"
        >
          AAP
        </label>
      </div>
      <div>
        <input
          id="Agriculture and Life Sciences"
          type="checkbox"
        />
        <label
          htmlFor="Agriculture and Life Sciences"
        >
          Agriculture and Life Sciences
        </label>
      </div>
      <div>
        <input
          id="Arts and Sciences"
          type="checkbox"
        />
        <label
          htmlFor="Arts and Sciences"
        >
          Arts and Sciences
        </label>
      </div>
      <div>
        <input
          id="Engineering"
          type="checkbox"
        />
        <label
          htmlFor="Engineering"
        >
          Engineering
        </label>
      </div>
      <div>
        <input
          id="Human Ecology"
          type="checkbox"
        />
        <label
          htmlFor="Human Ecology"
        >
          Human Ecology
        </label>
      </div>
      <div>
        <input
          id="ILR"
          type="checkbox"
        />
        <label
          htmlFor="ILR"
        >
          ILR
        </label>
      </div>
      <div>
        <input
          id="SC Johnson School of Business"
          type="checkbox"
        />
        <label
          htmlFor="SC Johnson School of Business"
        >
          SC Johnson School of Business
        </label>
      </div>
    </form>
  </div>
</div>
`;
